src/vector_calculus.f90

```diff
@@ -0,0 +1,368 @@
+module m_vector_calculus
+  use m_allocator, only: allocator_t, field_t
+  use m_base_backend, only: base_backend_t
+  use m_common, only: dp, DIR_X, DIR_Y, DIR_Z, &
+                      RDR_X2Y, RDR_X2Z, RDR_Y2X, RDR_Y2Z, RDR_Z2X, RDR_Z2Y
+  use m_tdsops, only: tdsops_t
+
+  implicit none
+
+  type :: vector_calculus_t
+    !! Defines vector calculus operators
+    class(base_backend_t), pointer :: backend
```

Comment:
This is the kind of circular pointer that I think we should try and avoid (i.e. a pointer to the `backend` object in the parent class of `vector_calculus`). Could we pass `backend` as an argument to `vector_calculus` methods? e.g.
```
call vector_calculus%curl(..., backend)
```

Reply:
We don't really have a circular pointer or dependency here. `backend`, `solver` and `vector_calculus` are all independent classes and there is no inheritance relationship among them. There are other classes too that use `backend`, like `time_integrator` and new wind turbine stuff. In a class diagram `backend` would be under all of them so I don't see a problem having a pointer to `backend` in these classes. Why do you think we should avoid?